main/state/game.lua

```
@@ -1,69 +1,32 @@
--- Throws an error if the specified property does not exist.
--- @param property The dot-separated path to the property to retrieve (e.g., "player.max_energy").
--- @return The value of the specified property, or nil if the property does not exist.
-function STATE_M.get_state_prop(property) 
-	local levels = {}
-	for level in string.gmatch(property, "[^.]+") do
-		table.insert(levels, level)
-	end
-
-	local result = STATE_M
-	for _, level in ipairs(levels) do
-		result = result[level]
-		if result == nil then
-			error('GAME STATE: Property is not set')
-			return nil
-		end
-	end
-	return result
-end
-
---- Updates or creates a nested property in STATE_M.
--- @param property The dot-separated path to the property (e.g., "player.max_energy").
--- @param newValue The value to set at the property path.
--- @param force If true, creates missing path segments; if false, throws error on missing paths.
-function STATE_M.set_state_prop(property, newValue, force)
-	local levels = {}
-	for level in string.gmatch(property, "[^.]+") do
-		table.insert(levels, level)
-	end
-
-	local result = STATE_M
-	for i = 1, #levels - 1 do
-		local level = levels[i]
-		if result[level] == nil then
-			if force then
-				result[level] = {}
-			else
-				error("GAME STATE: Property path does not exist: " .. table.concat(levels, ".", 1, i))
-			end
-		end
-		result = result[level]
-	end
+local game_state = {
+	player_max_energy = 5,
+	player_max_speed = 150,
+	player_max_health = 100,
+	player_coins = 50,
+	inventory_size_row = 2,
+	inventory_size_col = 4
+}
 
-	local finalLevel = levels[#levels]
-	if finalLevel then
-		if not force and result[finalLevel] == nil then
-			error("GAME STATE: Property does not exist: " .. property)
-		end
-		result[finalLevel] = newValue
+function game_state.get_state_property(property)
+	local property = game_state[property]
+	if property == nil then
+		print("Error: Property '" .. tostring(property) .. "' not found in game state.")
```

Also not sure if you want to throw an error here. We might want to have properties that are nil. Probably best to just return `game_state[property]`